src/index.js

```diff
@@ -259,7 +261,13 @@ class FsDatastore {
    * @returns {PullStream}
    */
   query (q /* : Query<Buffer> */) /* : QueryResult<Buffer> */ {
-    let tasks = [glob(path.join(this.path, '**', '*' + this.opts.extension))]
+    // glob expects a POSIX path
+    let prefix = q.prefix || '**'
+    let pattern =
+      path.join(this.path, prefix, '*' + this.opts.extension)
+      .split(path.sep)
+      .join('/')
```

Comment:
Using a regex to do a replace should be more efficient than splitting into an array and rejoining. I.e. 

```
// at the top of the file
const reg = new RegExp(path.sep , 'g')

path.join(..).replace(reg, '/')
```

Reply:
Are you sure?  My feeling is that regex is very expensive, compared to a few bytes (128) on the heap.

Reply:
Well `split` + `join` means you are allocating a new array and a new string, vs if you use a regex it only allocates a new string for the replacement.

Reply:
and if you are unlucky the code path for `.split` could end up allocating new strings for each part, so array + x string allocations for the individual parts

Reply:
Yeah but regex needs to do parsing and then build a state table. 

This is about to turn into a religious war that I want to avoid.  So lets kiss and makeup.

Reply:
fair enough, lets worry about once it becomes an actual perf issue